server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -86,29 +86,46 @@
     private final MIMEMessageConverter mimeMessageConverter;
     private final MailSpool mailSpool;
     private final MailFactory<Id> mailFactory;
+    private final ThrowingFunction<MailboxSession, Optional<Mailbox<Id>>> outboxProvider;
+    private final ThrowingFunction<MailboxSession, Optional<Id>> draftsIdProvider;
 
-    @Inject
     @VisibleForTesting
+    @Inject
     SetMessagesCreationProcessor(MailboxMapperFactory<Id> mailboxMapperFactory,
                                  MailboxManager mailboxManager,
                                  MailboxSessionMapperFactory<Id> mailboxSessionMapperFactory,
                                  MIMEMessageConverter mimeMessageConverter,
                                  MailSpool mailSpool,
                                  MailFactory<Id> mailFactory) {
+        this(mailboxMapperFactory, mailboxManager, mailboxSessionMapperFactory, mimeMessageConverter, mailSpool, mailFactory, null, null);
+    }
+
+    @VisibleForTesting
+    SetMessagesCreationProcessor(MailboxMapperFactory<Id> mailboxMapperFactory,
+                                 MailboxManager mailboxManager,
+                                 MailboxSessionMapperFactory<Id> mailboxSessionMapperFactory,
+                                 MIMEMessageConverter mimeMessageConverter,
+                                 MailSpool mailSpool,
+                                 MailFactory<Id> mailFactory,
+                                 ThrowingFunction<MailboxSession, Optional<Mailbox<Id>>> outboxProvider,
+                                 ThrowingFunction<MailboxSession, Optional<Id>> draftsIdProvider) {
+
         this.mailboxMapperFactory = mailboxMapperFactory;
         this.mailboxManager = mailboxManager;
         this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
         this.mimeMessageConverter = mimeMessageConverter;
         this.mailSpool = mailSpool;
         this.mailFactory = mailFactory;
+        this.outboxProvider = outboxProvider == null ? this::getOutbox : outboxProvider;
+        this.draftsIdProvider = draftsIdProvider == null ? this::getDraftsId : draftsIdProvider;
```

Comment:
mailboxManager has (too) many responsibilities. That would make implementation and testing code too brittle.
Adding a "test ctor" with injection, in the meantime, helps with identifying used features that might become candidates for refactoring, without (hard) coupling now with mailboxManager.